lib/auth/tests.py

```diff
@@ -4,341 +4,125 @@
-        self.assertEqual(user.email, self.fake_user[0][1]['mail'][0])
-
-        user_saved = User.objects.get(email='peterbe@mozilla.com')
-        self.assertEqual(user_saved.first_name, user.first_name)
-        self.assertEqual(user_saved.last_name, user.last_name)
-
-        self.assertTrue(user_saved.groups.filter(name='Localizers').exists())
-
-    def test_authenticate_with_ldap_wrong_password(self):
-        ldap.initialize = Mock(return_value=MockLDAP({
-          'dc=mozilla': self.fake_user,
-          'ou=groups,dc=mozilla': self.fake_group
-        }, credentials={
-          self.fake_user[0][0]: 'rightsecret'
-        }))
-        backend = MozLdapBackend()
-        user = backend.authenticate('foo', 'secret')
-        self.assertIsNone(user)
-
-    def test_authenticate_with_ldap_wrong_username(self):
-        ldap.initialize = Mock(return_value=MockLDAP({
-          'dc=mozilla': self.fake_user,
-          'ou=groups,dc=mozilla': self.fake_group
-        }, credentials={
-          'some-other-uid': 'secret'
-        }))
-        backend = MozLdapBackend()
-        user = backend.authenticate('foo', 'secret')
-        self.assertIsNone(user)
-
-    def test_authentication_ldap_username_not_recognized(self):
-        ldap.initialize = Mock(return_value=MockLDAP({
-          'dc=mozilla': None,
-          'ou=groups,dc=mozilla': self.fake_group
-        }, credentials={
-          self.fake_user[0][0]: 'secret'
-        }))
-        backend = MozLdapBackend()
-        user = backend.authenticate('foo', 'secret')
-        self.assertFalse(user)
-
-    def test_ldap_server_down_error(self):
-        ldap.initialize = Mock(return_value=MockLDAP({
-          'dc=mozilla': None,
-          'ou=groups,dc=mozilla': self.fake_group
-        }, credentials={
-          self.fake_user[0][0]: 'secret'
-        }))
-        backend = BreakingMozLdapBackend()
-        with self.assertRaises(ldap.SERVER_DOWN):
-            backend._authenticate_ldap('foo@example.com', 'secret')
-
-        # try it from the "outside"
-        from django.core.urlresolvers import reverse
-        url = reverse('login')
-
-        with self.settings(
-            AUTHENTICATION_BACKENDS = (
-              'lib.auth.tests.BreakingMozLdapBackend',
+from django.contrib.auth.models import Group
+
+from .backends import ElmoOIDCBackend
+
+
+class OIDCBackendTestCase(TestCase):
+    def test_create(self):
+        with patch.object(
+            ElmoOIDCBackend,
+            'update_user',
+            side_effect=lambda user, claims: user
+        ) as update_mock:
+            backend = ElmoOIDCBackend()
+            claims = {
+                'given_name': 'Axel',
+                'family_name': 'Hecht',
+                'email': 'axel@example.com'
+            }
+            user = backend.create_user(claims)
+            update_mock.assert_called_once_with(user, claims)
+
+    def test_no_update(self):
```

Comment:
Another approach would be to also mock ``update_user`` and assert that its never called.

Reply:
I'm not sure what you meant here. That said I'm doing a couple of changes to the test here to make it easier to read.